src/goto-instrument/contracts/contracts.cpp

```diff
@@ -678,6 +685,12 @@ void code_contractst::apply_loop_contract(
   local_may_aliast local_may_alias(goto_function);
   natural_loops_mutablet natural_loops(goto_function.body);
 
+  if(!natural_loops.loop_map.size())
+    return;
+
+  goto_function_inline(
+    goto_functions, function_name, ns, log.get_message_handler());
+
```

Comment:
inlining after loop detection will miss loops that are hidden behind function calls that do not have contracts

Reply:
which is okay? because they don't have contracts anyway?

Reply:
I don't object but do note that this kind of inlining can get *expensive*.